src/sidebar/components/SidebarTabbedPanel.tsx

```diff
@@ -15,6 +15,11 @@ import { applyTheme } from '../helpers/theme';
 import { withServices } from '../service-context';
 import type { AnnotationsService } from '../services/annotations';
 import { useSidebarStore } from '../store';
+import ThreadList from './ThreadList';
+import { useRootThread } from './hooks/use-root-thread';
+
+const idForTab = (name: TabName) => `${name}-tab`;
+const idForPanel = (name: TabName) => `${name}-panel`;
```

Comment:
This works, as long as there is only one instance on the page. If there were multiple, we'd need to generate unique IDs locally.

Reply:
At one point this used randomly generated IDs with `useId()`, instead of tab names.

However, in order to determine the selected one, it required an ugly logic to match the proper id from `selectedTab`.

Because of that I ended up just using tab names and `selectedTab` to determine the selected tab's id.